### PR TITLE
dragonball: metrics support for virtio-mmio device

### PR DESCRIPTION
**Pull request by lisongqian**
This PR adds metrics support for virtio-mmio device.

Fixes: #7248

Signed-off-by: Songqian Li <mail@lisongqian.cn>

### CODE REVIEW

File: src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs [hunk 1]
```diff
@@ -25,6 +27,24 @@ const DEVICE_STATUS_DRIVER: u32 = DEVICE_STATUS_ACKNOWLEDE | DEVICE_DRIVER;
...
+    /// Number of times when handling read events on a mmio device.
+    pub read_count: SharedIncMetric,
+    /// Number of times when handling write events on a mmio device.
+    pub write_count: SharedIncMetric,
+    /// Number of times when handling read events on a mmio device failed.
+    pub read_fails: SharedIncMetric,
+    /// Number of times when handling write events on a mmio device failed.
+    pub write_fails: SharedIncMetric,
+    /// Number of times when activated on a mmio deivce.
+    pub activate_count: SharedIncMetric,
```
Comment (ZizhengBian) [inline src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs:44]: The activate action only need to do once, we don't need to add this metric

Comment (ZizhengBian) [reply src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs:44]: Same as other devices

Comment (ZizhengBian) [reply src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs:44]: you can use an atomic variable to represent not activated, activated success or fail,

File: src/dragonball/src/metric.rs [hunk 2]
```diff
@@ -77,6 +78,8 @@ pub struct DragonballMetrics {
     pub rtc: Arc<RTCDeviceMetrics>,
     /// Metrics related to serial device.
     pub serial: HashMap<String, Arc<SerialDeviceMetrics>>,
+    /// Metrics related to a MMIO transport.
+    pub mmio: HashMap<String, Arc<MMIODeviceMetrics>>,
```
Comment (ZizhengBian) [inline src/dragonball/src/metric.rs:82]: It's better to add mmio transport layer's metric to each device's metric

File: src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs (hunk 1 above)
Comment (lisongqian) [reply src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs:44]: Considering the possibility of multiple failed activations, I retained the `activate_fails`. And I used `activate_flag` to represent the state.
- If `activate_flag` is 1, it means the device is activated.
- If `activate_flag` is 0 and `activate_fails` is 0, it means the device is not activated.
- If `activate_flag` is 0 and `activate_fails` is not 0, it means the device has failed to activate.

File: src/dragonball/src/metric.rs (hunk 2 above)
Comment (lisongqian) [reply src/dragonball/src/metric.rs:82]: I agree with you. I will consider it later.

File: src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs (hunk 1 above)
Comment (ZizhengBian) [reply src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs]: Great job!

**Review by ZizhengBian — APPROVED** [review]
LGTM

**Pull request closed without merging**